modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1893,6 +1894,12 @@ def apply_full_axis(
             The data types of the result. This is an optimization
             because there are functions that always result in a particular data
             type, and allows us to avoid (re)computing it.
+        sync_axes : bool, default: True
```

Comment:
maybe check_axes_sync?  make it obvious that setting to False means skipping a check?